src/hotspot/share/tsan/tsanOopMapTable.cpp

```diff
@@ -70,15 +69,23 @@ TsanOopMapTable::~TsanOopMapTable() {
   clear();
 }
 
+bool TsanOopMapTable::add_entry(TsanOopMapTableKey *entry, size_t size) {
+  assert(TsanOopMap_lock->is_locked(), "sanity check");
```

Comment:
If we get rid of the locking in `WeakProcessor::Task::work()`, this assert could be `SafepointSynchronize::is_at_safepoint()` instead.

Reply:
Please see above comment.